### PR TITLE
Extra attributes for css/js file elements

### LINKED ISSUE #5604: More attributes option in loadCSSFile() and loadJavaScriptFile()
You should be able to load an external files with extra attributes like "crossorigin" and/or "integrity"

### PR DESCRIPTION
**Pull request by SychO9**
Closes #5604

### CODE REVIEW

File: Sources/Subs.php [hunk 1]
```diff
@@ -3894,8 +3894,16 @@ function template_javascript($do_deferred = false)
...
 
 			else
+			{
 				echo '
-	<script src="', $js_file['fileUrl'], isset($file['options']['seed']) ? $file['options']['seed'] : '', '"', !empty($js_file['options']['async']) ? ' async' : '', !empty($js_file['options']['defer']) ? ' defer' : '', '></script>';
+	<script src="', $js_file['fileUrl'], isset($js_file['options']['seed']) ? $js_file['options']['seed'] : '', '"', !empty($js_file['options']['async']) ? ' async' : '', !empty($js_file['options']['defer']) ? ' defer' : '';
+
+				if (!empty($js_file['options']['attributes']))
+					foreach ($js_file['options']['attributes'] as $key => $value)
+						echo ' ', $key, '="', $value, '"';
```
Comment (Sesquipedalian) [inline Sources/Subs.php]: For better HTML5 output, you'll want to handle boolean values differently. This should do the job:

```
foreach ($js_file['options']['attributes'] as $key => $value)
{
	if (is_bool($value))
		echo !empty($value) ? ' ' . $key : '';
	else
		echo ' ', $key, '="', $value, '"';
}
```

File: Sources/Subs.php [hunk 2]
```diff
@@ -3997,8 +4008,16 @@ function template_css()
...
 	if (!empty($normal))
 		foreach ($normal as $nf)
+		{
 			echo '
-	<link rel="stylesheet" href="', $nf, '">';
+	<link rel="stylesheet" href="', $nf['url'], '"';
+
+			if (!empty($nf['attributes']))
+				foreach ($nf['attributes'] as $key => $value)
+					echo ' ', $key, '="', $value, '"';
```
Comment (Sesquipedalian) [inline Sources/Subs.php]: See previous comment for advice on how to handle boolean attributes.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]

**Review by Sesquipedalian — APPROVED** [review]

**Comment by umairaziz1** [comment]
Bug

**Comment by Sesquipedalian** [comment]
> Bug

Can you explain what you mean a little more fully, @umairaziz1?

**Review by umairaziz1 — APPROVED** [review]

**Pull request merged**